osu.Server.Queues.ScoreStatisticsProcessor/Commands/Performance/PerformanceCommand.cs

```diff
@@ -69,6 +73,37 @@ await ProcessPartitioned(userIds, async (db, transaction, userId) =>
                 if (userStats == null)
                     return;
 
+                // Mania per-key ranking statistic updates.
+                if (RulesetId == 3)
+                {
+                    await updateKeyStats(4);
+                    await updateKeyStats(7);
+
+                    async Task updateKeyStats(int keyCount)
+                    {
+                        string keyCountTableName = $"osu_user_stats_mania_{keyCount}k";
+                        var keyModeStats = db.QueryFirstOrDefault<UserStatsManiaKeyCount>($"SELECT * FROM {keyCountTableName} WHERE `user_id` = @user_id", userStats, transaction);
+
+                        if (keyModeStats != null)
+                        {
+                            double rankScoreBefore = keyModeStats.rank_score;
+                            double accBefore = keyModeStats.accuracy_new;
+
+                            await ManiaKeyModeProcessor.UpdateUserStatsAsync(keyModeStats, keyCount, db, transaction, updateIndex: false);
+
+                            if (Math.Abs(rankScoreBefore - keyModeStats.rank_score) > 0.1 ||
+                                Math.Abs(accBefore - keyModeStats.accuracy_new) > 0.1)
+                            {
+                                await db.ExecuteAsync(
+                                    $"UPDATE `{keyCountTableName}` "
+                                    + $"SET `rank_score` = @rank_score, `playcount` = @playcount, `rank_score_index` = @rank_score_index, `accuracy_new` = @accuracy_new, "
```

Comment:
Why is `rank_score_index` in this update if the `UpdateUserStatsAsync()` call prior was with `updateIndex: false`, which means that in theory the rank wouldn't have been touched...?

And, as a corollary to the above, what will ensure that `rank_score_index` will be eventually correct after this is run? The daily rank calculation cron I guess?

Reply:
> And, as a corollary to the above, what will ensure that rank_score_index will be eventually correct after this is run? The daily rank calculation cron I guess?

That's the one, yep. I'm doing it this way in line with the (general) user totals update command.